src/txn/txn_recover.c

```diff
@@ -503,6 +504,23 @@ __recovery_set_ckpt_base_write_gen(WT_RECOVERY *r)
     return (ret);
 }
 
+/*
+ * __recovery_txn_setup_initial_state --
+ *     Setup the transaction initial state required by rollback to stable.
+ */
+static int
+__recovery_txn_setup_initial_state(WT_SESSION_IMPL *session, WT_RECOVERY *r)
+{
+    WT_RET(__recovery_set_checkpoint_timestamp(r));
+    WT_RET(__recovery_set_oldest_timestamp(r));
```

Comment:
What about setting the stable timestamp ?

Reply:
Yes, it was set outside of the function. Now I moved it inside this function itself to be clear.

Reply:
This seems like a good question - I wish I knew the answer :) It used to be that stable was implied to be oldest, if not set explicitly, but I don't know if we can rely on that any longer.

Reply:
Not anymore. we need to set them explicitly. The `pinned timestamp` is calculated purely based on the `oldest timestamp`. We treat the oldest timestamp as 0 if we don't set it.

Reply:
lgtm now - thanks for taking the time to explain and update the code Hari